Ensure string values are null terminated when writing them

Numpsy opened the pull request:
refs #47 

When testing adding or modifying properties, I noticed that new or modified string properties weren't getting null terminated, and that was causing issues with office apps like powerpoint which were showing truncated property values.

I went to add the null values, but it isn't quite that simple as any properties that OpenMxdf has read itself will retain the null terminators from the originating file whereas new values supplied by a user generally won't, so - try to check if the value has a trailing null already and only write an extra one if not.

This *seems* to work, though i'm not 100% sure on the situation with empty strings (The Windows native compound doc apis seem to set a length of 1 and include a null for empty string properties, so I've done the same)

ironfede reviewed (changes requested):
@Numpsy could you please add a simple unit test for null terminated strings in #102 ?
Many thanks!
Federico

Numpsy commented:
Added a couple of tests, and also changed to check for the encoded byte array ending with a null instead of the input string, as it seems that
```csharp
anyString.EndsWith("\0")
```
is always true in .NET 6

ironfede reviewed (approved):
Ok, many thanks @Numpsy
